### PR TITLE
[RORO-4674][JT] remove warnings

### PR DESCRIPTION
**Pull request by tianjianfeng**

### CODE REVIEW

File: app/uk/gov/hmrc/gvmshaulierstatus/repositories/HaulierStatusRepository.scala [hunk 1]
```diff
@@ -68,9 +68,8 @@ class HaulierStatusRepository @Inject() (
         .toFuture()
     )
 
-  def findAndUpdate(correlationId: CorrelationId, status: Status)(implicit
-    instant: Instant = Instant.now(Clock.systemUTC())
-  ): Future[Option[String]] =
+  def findAndUpdate(correlationId: CorrelationId, status: Status): Future[Option[String]] = {
```
Comment (tipperba) [inline app/uk/gov/hmrc/gvmshaulierstatus/repositories/HaulierStatusRepository.scala]: If we're removing this parameter in both these defs just to get rid of a wartremover `DefaultArguments` warning, I think we're best off suppressing it instead. Looks like the parameter has been added to allow unit testing to confirm the respective date/time field is being updated as expected.

Comment (tipperba) [reply app/uk/gov/hmrc/gvmshaulierstatus/repositories/HaulierStatusRepository.scala]: ...and if we do want to decide to get rid of the parameter, we presumably don't need to make the val implicit and probably don't need a val at all. 

Think I'd be in favour of suppressing this particular warning though and keeping the test.

File: app/uk/gov/hmrc/gvmshaulierstatus/services/HaulierStatusService.scala [hunk 2]
```diff
@@ -66,14 +66,14 @@ class HaulierStatusService @Inject() (
 
   def updateStatus()(implicit headerCarrier: HeaderCarrier): Future[Unit] =
     haulierStatusRepository.findAllOlderThan(appConfig.intervalSeconds, appConfig.limit).flatMap { documents =>
-      val receivedDocsPercentage = if (documents.nonEmpty) (documents.count(_.status == Received).toFloat / documents.size) * 100 else 0
+      val receivedDocsPercentage = if (documents.nonEmpty) (documents.count(_.status.value === Received.value).toFloat / documents.size) * 100 else 0
```
Comment (tipperba) [inline app/uk/gov/hmrc/gvmshaulierstatus/services/HaulierStatusService.scala]: This looks like a big step backwards - we don't want to replace a type safe match with a String match.

File: app/uk/gov/hmrc/gvmshaulierstatus/services/HaulierStatusService.scala [hunk 3]
```diff
@@ -66,14 +66,14 @@ class HaulierStatusService @Inject() (
...
       receivedPercentages.add(receivedDocsPercentage)
       logger.debug(s"% of documents with 'Received' status: ${String.format("%.2f", receivedDocsPercentage)}")
 
-      if (documents.isEmpty || (currentState == UNAVAILABLE && receivedPercentages.forAllAndFull(_ >= appConfig.orangeThreshold))) {
+      if (documents.isEmpty || (currentState.value === UNAVAILABLE.value && receivedPercentages.forAllAndFull(_ >= appConfig.orangeThreshold))) {
```
Comment (tipperba) [inline app/uk/gov/hmrc/gvmshaulierstatus/services/HaulierStatusService.scala]: same here

File: app/uk/gov/hmrc/gvmshaulierstatus/services/HaulierStatusService.scala [hunk 4]
```diff
@@ -66,14 +66,14 @@ class HaulierStatusService @Inject() (
...
         setState(AVAILABLE)
       } else if (
-        (currentState == AVAILABLE && receivedDocsPercentage < appConfig.redThreshold) || (currentState == AVAILABLE && receivedPercentages
+        (currentState.value === AVAILABLE.value && receivedDocsPercentage < appConfig.redThreshold) || (currentState.value === AVAILABLE.value && receivedPercentages
```
Comment (tipperba) [inline app/uk/gov/hmrc/gvmshaulierstatus/services/HaulierStatusService.scala]: ditto

File: app/uk/gov/hmrc/gvmshaulierstatus/services/HaulierStatusService.scala [hunk 5]
```diff
@@ -84,7 +84,7 @@ class HaulierStatusService @Inject() (
     }
 
   private def logMissingReceipts(documents: Seq[HaulierStatusDocument]): Unit = {
-    val createdDocs = documents.filter(_.status == Created)
+    val createdDocs = documents.filter(_.status.value === Created.value)
```
Comment (tipperba) [inline app/uk/gov/hmrc/gvmshaulierstatus/services/HaulierStatusService.scala]: and here

File: app/uk/gov/hmrc/gvmshaulierstatus/repositories/HaulierStatusRepository.scala (hunk 1 above)
Comment (tianjianfeng) [reply app/uk/gov/hmrc/gvmshaulierstatus/repositories/HaulierStatusRepository.scala]: done

File: app/uk/gov/hmrc/gvmshaulierstatus/services/HaulierStatusService.scala (hunk 2 above)
Comment (tianjianfeng) [reply app/uk/gov/hmrc/gvmshaulierstatus/services/HaulierStatusService.scala]: updated.

File: app/uk/gov/hmrc/gvmshaulierstatus/services/HaulierStatusService.scala (hunk 3 above)
Comment (tianjianfeng) [reply app/uk/gov/hmrc/gvmshaulierstatus/services/HaulierStatusService.scala]: updated.

File: app/uk/gov/hmrc/gvmshaulierstatus/services/HaulierStatusService.scala (hunk 4 above)
Comment (tianjianfeng) [reply app/uk/gov/hmrc/gvmshaulierstatus/services/HaulierStatusService.scala]: updated.

File: app/uk/gov/hmrc/gvmshaulierstatus/services/HaulierStatusService.scala (hunk 5 above)
Comment (tianjianfeng) [reply app/uk/gov/hmrc/gvmshaulierstatus/services/HaulierStatusService.scala]: updated.

**Review by tipperba — APPROVED** [review]

**Pull request merged**